terraform/environments/core-logging/s3_logging.tf

```diff
@@ -221,44 +221,33 @@ module "s3-bucket-cloudtrail-logging" {
 
   replication_enabled = true
   replication_region  = "eu-west-1"
-  versioning_enabled  = true
-
-  lifecycle_rule = [
-    {
-      id      = "main"
-      enabled = "Enabled"
-      prefix  = ""
-      tags    = {}
-      transition = [
-        {
-          days          = 90
-          storage_class = "STANDARD_IA"
-          }, {
-          days          = 365
-          storage_class = "GLACIER"
-        }
-      ]
-      expiration = {
-        days = 730
-      }
-      noncurrent_version_transition = [
-        {
-          days          = 90
-          storage_class = "STANDARD_IA"
-          }, {
-          days          = 365
-          storage_class = "GLACIER"
-        }
-      ]
-      noncurrent_version_expiration = {
-        days = 730
-      }
-    }
-  ]
 
   tags = local.tags
 }
 
+data "aws_iam_policy_document" "cloudtrail_logging_bucket_policy" {
+  statement {
+    sid       = "AllowS3ServerAccessLoggingPutObjectFromSourceBucketWithinAccount"
+    effect    = "Allow"
+    actions   = ["s3:PutObject"]
+    resources = ["${module.s3-bucket-cloudtrail-logging.bucket.arn}/*"]
+    principals {
+      type        = "Service"
+      identifiers = ["logging.s3.amazonaws.com"]
+    }
+    condition {
+      test     = "ArnEquals"
+      variable = "aws:SourceArn"
+      values   = [module.s3-bucket-cloudtrail-logging.bucket.arn]
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceAccount"
+      values   = [data.aws_caller_identity.current.account_id]
+    }
+  }
+}
+
```

Comment:
General Policy structure from [Grant Log Deliver Permissions via Bucker Policy - AWS Docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/enable-server-access-logging.html#grant-log-delivery-permissions-bucket-policy)

I changed `ArnLike` to `ArnEquals` - I thought it sounded clearer, but they both function the same ([ref](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html))